include/deal.II/lac/block_matrix_base.h

```diff
@@ -1350,8 +1350,8 @@ namespace BlockMatrixIterators
       {
         // we were asked to create the end
         // iterator for this matrix
-        this->row_block = numbers::invalid_size_type;
-        this->col_block = numbers::invalid_size_type;
+        this->row_block = numbers::invalid_unsigned_int;
+        this->col_block = numbers::invalid_unsigned_int;
```

Comment:
For the future: the changes in this file are logically unrelated to what you do in this pull request. Put unrelated changes into separate PRs. That makes it much easier to review and get things merged!